projects/plugins/jetpack/extensions/shared/use-autosave-and-redirect/index.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect, dispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+
+function redirect( url, callback ) {
+	if ( callback ) {
+		callback( url );
+	}
+	window.top.location.href = url;
+}
+
+export default function useAutosaveAndRedirect( redirectUrl, onRedirect = noop ) {
+	const [ isRedirecting, setIsRedirecting ] = useState( false );
+
+	const { isAutosaveablePost, isDirtyPost, currentPost } = useSelect( select => {
+		const editorSelector = select( 'core/editor' );
```

Comment:
Another totally optional style change is to destructure the object:

```
const { getCurrentPost, isEditedPostAutosaveable, isEditedPostDirty } = select( 'core/editor' );
```

Reply:
@ramonjd I _think_ calling these methods within the useSelect is preferred (I remember seeing a similar example in p7H4VZ-2Q6-p2) so that changes in the value returned by the selector causes a re-render.

Reply:
That makes sense. Thanks for the context @andrewserong 👍